js.go

```diff
@@ -2964,6 +2988,11 @@ func (sub *Subscription) FetchBatch(batch int, opts ...PullOpt) (MessageBatch, e
 			usrMsg, err = checkMsg(msg, true, false)
 			if err != nil {
 				if err == ErrTimeout {
+					subjectParts := strings.Split(msg.Subject, ".")
```

Comment:
If someone uses Pull with MaxBytes set, and request timeouts on the client before getting all messages, we could get `NATS/1.0 409 Batch Completed` that will be added in 2.10.0. We should probably handle and discard that error too.

Reply:
I'll add that in a separate PR, together with tests.